src/GraphQL/Types/Scalars/TimeSpanMillisecondsGraphType.cs

```diff
@@ -22,15 +22,14 @@ public TimeSpanMillisecondsGraphType()
         public override object Serialize(object value) => value switch
         {
             TimeSpan timeSpan => (long)timeSpan.TotalMilliseconds,
-            int i => i,
-            long l => l,
+            int _ => value,
```

Comment:
```suggestion
            int i => (long)i,
```

Bug was here. Serialize should always return the same type. In that case `long`. You may check other scalars to verify.

Reply:
Fixed.  Will review again after ValueConverter is removed.

Reply:
ok